frame/nomination-pools/src/lib.rs

```diff
@@ -582,19 +582,200 @@ pub struct PoolRoles<AccountId> {
 	pub state_toggler: Option<AccountId>,
 }
 
+/// Pool commission.
+///
+/// The pool `root` can set commission configuration after pool creation. By default, all commission
+/// values are `None`. Pool `root`s can set `max` and `throttle` configurations before setting an
+/// initial commission `current` value - the commission percentage and payee of commission.
+///
+/// The `max` commission value can only be decreased after the initial value is
+/// set, to prevent commission from repeatedly increasing.
+///
+/// An optional commission throttle allows the pool to set strict limits to how much
+/// commission can change in each update, and how often updates can take place.
+/// If a `throttle` is set before a commission `current` is set, the initial commission `current`
+/// value will not be subject to throttling. Subsequent commission updates will be.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Commission<T: Config> {
+	/// Optional commission rate of the pool along with the account commission is paid to.
+	pub current: Option<(Perbill, T::AccountId)>,
+	/// Optional maximum commission that can be set by the pool `root`. Once set, this value
+	/// can only be updated to a decreased value.
+	pub max: Option<Perbill>,
+	/// Optional configiration around how often commission can be updated, and when the last
+	/// commission update took place.
+	pub throttle: Option<CommissionThrottle<T>>,
+}
+
+impl<T: Config> Default for Commission<T> {
+	fn default() -> Self {
+		Self { current: None, max: None, throttle: None }
+	}
+}
+
+impl<T: Config> Commission<T> {
+	/// Get the current commission percentage of this pool.
+	/// Returns zero if commission has not yet been configured.
+	fn commission_or_zero(&self) -> Perbill {
+		self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero())
+	}
+
+	/// Returns true if a commission percentage updating to `to` would exhaust the throttle limit.
+	///
+	/// A commission update will be throttled (disallowed) if:
+	/// 1. not enough blocks have passed since the previous commission update took place, and
+	/// 2. the new commission is larger than the maximum allowed increase.
+	///
+	/// Throttling is not applied to commission updates if `current` is still `None`.
+	fn throttling(&self, to: &Perbill) -> bool {
+		let throttle = self.throttle.as_ref().map(|t| t).or(None);
+		if let Some(t) = throttle {
+			// check for `min_delay` throttling
+			if t.previous_set_at.map_or(false, |p| {
+				<frame_system::Pallet<T>>::block_number().saturating_sub(p) <
+					t.change_rate.min_delay
+			}) {
+				return true
+			}
+			// check for `max_increase` throttling
+			(*to).saturating_sub(self.commission_or_zero()) > t.change_rate.max_increase &&
+				self.current.is_some()
```

Comment:
It is checked because you can only throttle if an existing commission exists. That said, it is trivial to move `current` outside as `throttling()` is only checked once in `maybe_update_current`, so `current` is now checked there.